tools/internal/autoupdate/githublatestrelease.go

```diff
@@ -5,31 +5,90 @@ import (
 	"errors"
 	"fmt"
 
-	"github.com/rancher/image-mirror/internal/config"
-
+	"github.com/Masterminds/semver/v3"
 	"github.com/google/go-github/v71/github"
+	"github.com/rancher/image-mirror/internal/config"
 )
 
-// GithubLatestRelease retrieves the tag of the latest release
-// of the configured github repository, and returns the configured
-// Images with this tag. It assumes that the images have the same
-// tags as the github releases.
-type GithubLatestRelease struct {
-	Owner      string
-	Repository string
-	Images     []AutoupdateImageRef
+// GithubRelease retrieves the tags of all github releases that match the VersionConstraint if
+// LatestOnly is false, or the tag of the latest release if LatestOnly is true.
+// It returns the configured Images with this tag.
+// It assumes that the images have the same tags as the github releases.
+type GithubRelease struct {
+	Owner             string
+	Repository        string
+	Images            []AutoupdateImageRef
+	LatestOnly        bool
+	VersionConstraint string
```

Comment:
IMO we should cut down on visual noise by adding `omitempty` to both `LatestOnly` and `VersionConstraint`.